src/policy/policy.cpp

```diff
@@ -120,16 +120,11 @@ bool AreInputsStandard(const CTransaction& tx, const CCoinsViewCache& mapInputs)
 
     for (unsigned int i = 0; i < tx.vin.size(); i++)
     {
-        const CTxOut& prev = mapInputs.GetOutputFor(tx.vin[i]);
-
-        if (prev.scriptPubKey.IsWithdrawLock()) {
-            if (!tx.vin[i].scriptSig.IsWithdrawProof()) {
-                if (tx.vout.size() < i)
-                    if (!tx.vout[i].nValue.IsExplicit() || tx.vout[i].nValue.GetAmount() > MAX_MONEY / 100)
-                        return false;
-            }
+        if (tx.vin[i].m_is_pegin) {
+            // This deals with p2sh in general only
             continue;
```

Comment:
this is probably also something to look out for. non-peg-in inputs are validated to some degree, whereas peg-ins are not validated at all. It probably makes sense to ensure that no caller of this function relies all inputs being validated.

Reply:
Yes, the trickiest part(versus previous implementation) is making sure that we are handling these cases correctly.